config_utilities/include/config_utilities/internal/visitor_impl.hpp

```diff
@@ -320,16 +363,26 @@ void Visitor::visitField(OrderedMap<K, ConfigT>& config, const std::string& fiel
     }
   }
 
-  if (visitor.mode == Visitor::Mode::kGet) {
+  if (visitor.mode == Visitor::Mode::kGet || visitor.mode == Visitor::Mode::kGetInfo) {
     const std::string name_space = joinNamespace(visitor.name_space, field_name);
     YAML::Node map_node(YAML::NodeType::Map);
     for (auto&& [key, sub_config] : config) {
-      visitor.data.sub_configs.emplace_back(getValues(sub_config, false, name_space, field_name));
+      if (visitor.mode == Visitor::Mode::kGetInfo) {
+        visitor.data.sub_configs.emplace_back(getInfo(sub_config, name_space, field_name));
+      } else {
+        visitor.data.sub_configs.emplace_back(getValues(sub_config, false, name_space, field_name));
+      }
       MetaData& new_data = visitor.data.sub_configs.back();
       map_node[key] = YAML::Clone(lookupNamespace(new_data.data, name_space));
       new_data.map_config_key = YAML::Node(key).as<std::string>();
     }
 
+    if (visitor.mode == Visitor::Mode::kGetInfo && config.empty()) {
+      // When getting info for empty maps still show them.
+      // TODO(lschmid): Implement.
+      map_node = YAML::Node("TEEEEEEEEEEEST");
```

Comment:
It would be helpful for this to be either an exception or a log message (I think I would be surprised if I had this show up on the gui side :smile:)

Reply:
Funnily enough this actually never shows up 😄 But good point!